aviary/subsystems/aerodynamics/gasp_based/interference.py

```diff
@@ -0,0 +1,377 @@
+                Aircraft.Wing.CENTER_DISTANCE,
+            ])
+        self.declare_partials(
+            'drag_loss_due_to_shielded_wing_area', [
+                'CROOT',
+                'CBODYWF',
+                'WBODYWF',
+            ])
+
+    def compute_partials(self, inputs, J):
+        WBODYWF, CROOT, TCBODYWF, CBODYWF, ZW_RF, SWF, XWQLF = inputs.values()
+
+        KVWF = .0194 - 0.14817*ZW_RF + 1.3515*ZW_RF**2
+        dKVWF_dZWRF = -0.14817 + 2*1.3515*ZW_RF
+        KLWF = 0.13077 + 1.9791*XWQLF + 3.3325*XWQLF**2 - \
+            10.095*XWQLF**3 + 4.7229*XWQLF**4
+        dKLWF_dXWQLF = 1.9791 + 2*3.3325*XWQLF - 3*10.095*XWQLF**2 + 4*4.7229*XWQLF**3
+        KDTWF = 0.73543 + .028571*SWF/(TCBODYWF*CBODYWF)
+        dKDTWF_dSWF = .028571/(TCBODYWF*CBODYWF)
+        # dKDTWF_dTCBODYWF = -.028571*SWF/(TCBODYWF*CBODYWF)**2
+        # dKDTWF_dCBODYWF = -.028571*SWF/(TCBODYWF*CBODYWF)**2
+
+        J['interference_independent_of_shielded_area', 'TCBODYWF'] = \
+            1.5*(TCBODYWF)*(CBODYWF**2)*KVWF*KLWF * \
+            (3*0.73543*TCBODYWF + 2*.028571*SWF/CBODYWF)
+        J['interference_independent_of_shielded_area', 'CBODYWF'] = \
+            1.5*(TCBODYWF**3)*KVWF*KLWF*(2*0.73543*CBODYWF + .028571*SWF/TCBODYWF)
+        J['interference_independent_of_shielded_area', 'ZW_RF'] = \
+            1.5*(TCBODYWF**3)*(CBODYWF**2)*dKVWF_dZWRF*KLWF*KDTWF
+        J['interference_independent_of_shielded_area', Aircraft.Wing.CENTER_DISTANCE] = \
+            1.5*(TCBODYWF**3)*(CBODYWF**2)*KVWF*dKLWF_dXWQLF*KDTWF
+        J['interference_independent_of_shielded_area', Aircraft.Fuselage.AVG_DIAMETER] = \
+            1.5*(TCBODYWF**3)*(CBODYWF**2)*KVWF*KLWF*dKDTWF_dSWF
+
+        J['drag_loss_due_to_shielded_wing_area', 'CROOT'] = 0.5*WBODYWF
+        J['drag_loss_due_to_shielded_wing_area', 'CBODYWF'] = 0.5*WBODYWF
+        J['drag_loss_due_to_shielded_wing_area', 'WBODYWF'] = 0.5*(CROOT + CBODYWF)
+
+
+class WingFuselageInterference_premission(om.Group):
+    """
+    This calculates an additional flat plate drag area due to general aerodynamic interference for wing-fuselage interference
+    (based on results from Hoerner's drag)
+    """
+
+    def setup(self):
+        self.add_subsystem('root_chord', root_chord(),
+                           promotes_inputs=['*'], promotes_outputs=['*'])
+        self.add_subsystem('common_variables', common_variables(),
+                           promotes_inputs=['*'], promotes_outputs=['*'])
+        self.add_subsystem('top_and_bottom_width', top_and_bottom_width(),
+                           promotes_inputs=['*'], promotes_outputs=['*'])
+        self.add_subsystem('body_ratios', body_ratios(),
+                           promotes_inputs=['*'], promotes_outputs=['*'])
+        self.add_subsystem('interference_drag', interference_drag(),
+                           promotes_inputs=['*'], promotes_outputs=['*'])
+
+
+class WingFuselageInterference_dynamic(om.ExplicitComponent):
```

Comment:
I assume "dynamic" is supposed to mean it is the mission component? If true, change name to `WingFuselageInterferenceMission`